docs/source/routing/performance/caching/response-cache.mdx

```diff
@@ -0,0 +1,513 @@
+minVersion: Router v1.40.0
+releaseStage: preview
+---
+
+<PlanRequired plans={["Free", "Developer", "Standard", "Enterprise"]}>
+
+Rate limits apply on the Free plan.
+Performance pricing applies on Developer and Standard plans.
+
+</PlanRequired>
+
+The GraphOS Router can cache GraphQL response data using PostgreSQL to improve query latency and reduce load on your subgraphs.
+
+## Overview
+
+Response caching enables the router to store and reuse GraphQL response data in a PostgreSQL database, reducing redundant subgraph requests and improving query performance. This distributed caching approach allows multiple router instances to share cached data, providing consistent performance across your infrastructure.
+
+Key features of response caching:
+
+- **Persistent storage**: Uses PostgreSQL for durable cache storage that survives router restarts
+- **Distributed by design**: Multiple router instances automatically share the same cache
+- **Advanced invalidation**: Sophisticated cache invalidation with HTTP endpoints and batch operations
+- **Enhanced observability**: Comprehensive metrics and tracing for cache performance monitoring
+- **Flexible configuration**: Per-subgraph configuration with fine-grained control over caching behavior
+
+### Benefits of response caching
+
+Response caching provides significant advantages for GraphQL applications:
+
+- **Improved performance**: Cached responses eliminate redundant subgraph requests, reducing query latency and improving user experience.
+- **Scalability**: Multiple router instances share the same cache, enabling horizontal scaling without cache duplication.
+- **Durability**: PostgreSQL persistence ensures cached data survives router restarts and infrastructure changes.
+- **Fine-grained control**: Configure different TTL values, invalidation rules, and private data handling per subgraph.
+- **Advanced invalidation**: Proactive cache invalidation through HTTP endpoints or subgraph response extensions keeps data fresh.
+
+For example, consider an e-commerce application where product information changes infrequently but user-specific cart data changes often. Response caching allows you to:
+
+- Cache product descriptions and prices with long TTL values (hours or days)
+- Cache user cart data with short TTL values (minutes)
+- Invalidate specific product caches when inventory or pricing changes
+- Share product caches across all users while maintaining user-specific cart isolation
+
+## Use response caching
+
+Follow this guide to enable and configure response caching in the GraphOS Router.
+
+### Prerequisites
+
+Response caching requires:
+
+- PostgreSQL database instance that your router can connect to (version 12 or later, recommended: 14+)
+- [GraphOS Enterprise plan](https://www.apollographql.com/pricing/) that [connects your router to GraphOS](/router/configuration/overview/#environment-variables)
+
+### Configure router for response caching
```

Comment:
We should start by introducing a very simple configuration without any optional configuration fields